src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -182,56 +188,48 @@ public void flush() {
    *
    * @return the number of items added to 'bundle'.
    */
-  public int drainNextBundleTo(Collection<? super E> outputCollection) {
+  public E takeBundle() {
     final Lock lock = this.lock;
     lock.lock();
     try {
-      Bundle outBundle = null;
+      E outBundle = null;
       if (closedBundles.size() > 0) {
         outBundle = closedBundles.remove(0);
```

Comment:
Good point. This is a small change and seems worth doing. We only need a `Queue`. Do you have a preference for the implementation? `ArrayDeque` and `LinkedList` both work.